hcl/hcl.go

```diff
@@ -39,6 +39,34 @@ const (
 	StackBlockType = "stack"
 )
 
+// OptionalCheck is a bool that can also have no configured value.
+type OptionalCheck int
+
+const (
+	// CheckIsUnset means no value was specified.
+	CheckIsUnset OptionalCheck = iota
+	// CheckIsTrue means the check is disabled.
+	CheckIsTrue
+	// CheckIsFalse means the check is enabled.
+	CheckIsFalse
+)
+
+// ValueOr returns if an OptionalCheck is enabled, or the given default if its unset.
+func (v OptionalCheck) ValueOr(def bool) bool {
+	if v == CheckIsUnset {
+		return def
+	}
+	return v == CheckIsTrue
+}
+
+// ToOptionalCheck creates an OptionalCheck value from a bool.
+func ToOptionalCheck(v bool) OptionalCheck {
+	if v {
+		return CheckIsTrue
+	}
+	return CheckIsFalse
+}
+
```

Comment:
That's a nice abstraction, I have the following questions:
* Would it make sense to rename the type to `OptionalBool`?
* Could we move it to a new package to make this reusable?
* Should we add unit tests for it?

Reply:
I also went through a couple of names, including `OptionalBool`. I settled for something less generic so I don't overpromise, because I didn't want to spend too much time on it for now.

Many other languages have a generic type for this, I considered using https://github.com/markphelps/optional, but this one also cannot be compared with `==`.